bodhi/server/buildsys.py

```diff
@@ -25,7 +26,23 @@
 _koji_hub = None
 
 
-class Buildsystem:
+def multicall_enabled(func):
+    '''Decorator to enable multicall handling for DevBuildsys methods'''
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        if not self.multicall:
+            return func(self, *args, **kwargs)
+
+        # disable multicall during execution, so that inner func calls to other
+        # methods don't append their results as well
+        self._multicall = False
+        result = func(self, *args, **kwargs)
+        self.multicall_result.append([result])
+        self._multicall = True
+    return wrapper
+
+
+class Buildsystem(object):
```

Comment:
Is this change needed for the goal at hand? If not, I suggest making it a separate pull request.